submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/model/SubmitWebRequest.kt

```diff
@@ -12,40 +15,45 @@ class OnBehalfRequest(
     private val userName: String?,
     private val register: Boolean?
 ) {
-    fun asRegisterRequest(): GetOrRegisterUserRequest {
-        return GetOrRegisterUserRequest(register.orFalse(), userEmail, userName)
-    }
+    fun asRegisterRequest(): GetOrRegisterUserRequest =
+        GetOrRegisterUserRequest(register.orFalse(), userEmail, userName)
 }
 
-@Suppress("LongParameterList")
-sealed class SubmitWebRequest(
-    val submitter: SecurityUser,
-    val onBehalfRequest: OnBehalfRequest?,
+data class SubmissionRequestParameters(
+    val fileMode: FileMode = COPY,
+    val preferredSources: List<PreferredSource> = emptyList(),
+    val attributes: List<ExtAttributeDetail> = emptyList(),
```

Comment:
Even this is convenient we should not expose our extended model (domain models) to user rest layer  as then we will need to evolve them indivudally.